spec/que/active_support/job_middleware_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+if defined?(::ActiveSupport)
+  require 'que/active_support/job_middleware'
+
+  describe Que::ActiveSupport::JobMiddleware do
+    let(:job) { Que::Job.new(**labels) }
+
+    let(:labels) do
+      {
+        job_class: "Foo",
+        priority: 100,
+        queue: "foo_queue",
+        latency: 100,
+      }
+    end
+
+    it "records metrics when job succeeds" do
+      called = false
+      subscriber = ::ActiveSupport::Notifications.subscribe("que_job.worked") do |message, started, finished, metric_labels|
+        assert_equal "que_job.worked", message
+        assert started != nil
+        assert finished != nil
+        assert_equal labels.merge(error: false), metric_labels
+        called = true
+      end
+
+      Que::ActiveSupport::JobMiddleware.call(job) { }
+
+      assert_equal true, called
+
+      ::ActiveSupport::Notifications.unsubscribe(subscriber)
+    end
+
+    it "records metrics when job fails" do
+      called = false
+      subscriber = ::ActiveSupport::Notifications.subscribe("que_job.worked") do |message, started, finished, metric_labels|
+        assert_equal "que_job.worked", message
+        assert started != nil
+        assert finished != nil
+        assert_equal labels.merge(error: true), metric_labels
+        called = true
+      end
+
+      Que::ActiveSupport::JobMiddleware.call(job) { job.que_error = "error" }
```

Comment:
Ah, I was thinking you'd need to raise within the block here, but you're right - an exception from a job does not bubble up to middleware.

Testing job failure is good, but I was meaning that you test that it's in an `ensure` - when it encounters an exception. If not caused by the job, perhaps by a subsequent middleware.

For realistic data, I think `job.que_error` should also be set to an instance of an exception, not that it would have any effect here.

Although.. I've just looked up the term `job_worked` in the code, and found that logging makes a distinction: `job_worked` vs `job_errored`. So maybe this middleware should do the same? Or find a more generic term - something like `job_attempted`?